src/yoroi-wallets/cardano/ShelleyWallet.ts

```diff
@@ -134,11 +142,21 @@ export class ShelleyWallet extends Wallet implements WalletInterface {
     this.setupSubscriptions()
     this.notify({type: 'initialize'})
 
+    this.initializeUtxoService()
+
     this.isInitialized = true
 
     return this.id
   }
 
+  initializeUtxoService() {
```

Comment:
it would be happening right before creating/opening the wallet.

Reply:
We need the `networkId` to instantiate the utxoService, and we have access to it when the wallet is created by the method `wallet.create`

Reply:
https://github.com/Emurgo/yoroi-mobile/blob/3383fe7c8cbea198e36265b5acac704efcafff92/src/yoroi-wallets/cardano/ShelleyWallet.ts#L93